pandas/core/frame.py

```diff
@@ -3256,14 +3256,20 @@ def _setitem_frame(self, key, value):
     def _set_item_frame_value(self, key, value: DataFrame) -> None:
         self._ensure_valid_index(value)
 
-        # align right-hand-side columns if self.columns
-        # is multi-index and self[key] is a sub-frame
-        if isinstance(self.columns, MultiIndex) and key in self.columns:
+        if key in self.columns:
             loc = self.columns.get_loc(key)
```

Comment:
can you add a comment: `# align columns ` before L3259

Reply:
Done